File: Allegro5 Test/goal.cpp
#include "goal.h"


goal::goal(void)
{
}

void goal::Init(ALLEGRO_BITMAP *image = 0, float x = 0, float y = 0, int boundX = 100, int boundY = 100, int collideType = GREY){
	GameObject::Init(x, y ,0,0,0,0,boundX, boundY);
	SetID(GOAL);
	SetAlive(true);
	SetCollidable(true);
	SetCollideType(collideType);

	frameWidth = al_get_bitmap_width(image);
	frameHeight = al_get_bitmap_height(image);

	if(image != 0){
		goal::image = image;
	}
}

void goal::Render(){
	GameObject::Render();

	al_draw_bitmap_region(image, 0, 0, frameWidth, frameHeight, x, y, 0);
}

goal::~goal(void)
{
}

File: js/goal.js
// Goal class - objectives for players to reach

class Goal extends GameObject {
    constructor() {
        super();
        this.setId(ID.GOAL);
        this.setCollidable(true);
    }
    
    init(texture, x, y, boundX, boundY, collideType) {
        super.init(x, y, 0, 0, 0, 0, boundX, boundY);
        this.setCollideType(collideType);
        
        if (texture) {
            this.sprite = new PIXI.Sprite(texture);
            this.sprite.x = x;
            this.sprite.y = y;
            this.sprite.width = boundX * 2;
            this.sprite.height = boundY * 2;
            this.sprite.anchor.set(0.5, 0.5);
            
            // Tint based on collision type
            if (collideType === COLLIDE_TYPE.WHITE) {
                this.sprite.tint = 0xFFFFFF; // White goal
            } else if (collideType === COLLIDE_TYPE.BLACK) {
                this.sprite.tint = 0x333333; // Black goal
            }
        }
    }
    
    update() {
        // Goals don't need to update
    }
    
    render() {
        super.render();
    }
}